libshaderc/src/shaderc_test.cc

```diff
@@ -232,6 +232,42 @@ TEST(CompileStringWithOptions, IfDefCompileOption) {
                                  shaderc_glsl_vertex_shader, options.get()));
 }
 
+TEST(CompileStringWithOptions, MessageRules) {
+  Compiler compiler;
+  compile_options_ptr options(shaderc_compile_options_initialize());
+  ASSERT_NE(nullptr, compiler.get_compiler_handle());
+
+  // Confirm that this shader only compiles with shaderc_msg_default;
+  // if using SPIR-V rules, glslang will fail to compile it
+  const std::string kGlslShader =
+    R"(#version 100
+       uniform highp sampler2D tex;
+       void main() {
+         gl_FragColor = texture2D(tex, vec2(0.0,0.0));
+       }
+  )";
+
+  EXPECT_FALSE(CompilesToValidSpv(compiler.get_compiler_handle(),
+                                  kGlslShader,
+                                  shaderc_glsl_fragment_shader, options.get()));
+
+  shaderc_compile_options_set_message_rules(options.get(), shaderc_msg_default);
+  EXPECT_TRUE(CompilesToValidSpv(compiler.get_compiler_handle(),
+                                 kGlslShader,
+                                 shaderc_glsl_fragment_shader, options.get()));
+
+  shaderc_compile_options_set_message_rules(options.get(), shaderc_msg_spirv_rules);
+  EXPECT_FALSE(CompilesToValidSpv(compiler.get_compiler_handle(),
+                                  kGlslShader,
+                                  shaderc_glsl_fragment_shader, options.get()));
+
+  shaderc_compile_options_set_message_rules(options.get(),
+                                            static_cast<shaderc_msg_rules>(shaderc_msg_spirv_rules | shaderc_msg_vulkan_rules));
```

Comment:
I don't like this `static_cast`, because the value it produces isn't a member of the enum.